code/go/0chain.net/smartcontract/dbs/event/user.go

```diff
@@ -48,6 +68,99 @@ func (edb *EventDb) addOrUpdateUsers(users []User) error {
 	}).Create(&users).Error
 }
 
+func (edb *EventDb) updateUserCollectedRewards(users []User) error {
+	var ids []string
+	var collectedRewards []int64
+	for _, u := range users {
+		ids = append(ids, u.UserID)
+		collectedRewards = append(collectedRewards, u.CollectedReward)
+	}
+
+	return CreateBuilder("users", "user_id", ids).
+		AddUpdate("collected_reward", collectedRewards, "users.collected_reward + t.collected_reward").Exec(edb).Error
+}
+
+func (edb *EventDb) updateUserTotalStake(dpls []DelegatePoolLock, shouldIncrease bool) error {
+	var ids []string
+	var stakes []int64
+	for _, dpl := range dpls {
+		ids = append(ids, dpl.Client)
+		if shouldIncrease {
+			stakes = append(stakes, dpl.Amount)
+			continue
+		}
+		stakes = append(stakes, -dpl.Amount)
+	}
+
+	return CreateBuilder("users", "user_id", ids).
+		AddUpdate("total_stake", stakes, "users.total_stake + t.total_stake").Exec(edb).Error
+}
+
+func (edb *EventDb) updateUserReadPoolTotal(rpls []ReadPoolLock, shouldIncrease bool) error {
+	var ids []string
+	var readpools []int64
+	for _, rpl := range rpls {
+		ids = append(ids, rpl.Client)
+		if shouldIncrease {
+			readpools = append(readpools, rpl.Amount)
+			continue
+		}
+		readpools = append(readpools, -rpl.Amount)
+
+	}
+	return CreateBuilder("users", "user_id", ids).
+		AddUpdate("read_pool_total", readpools, "users.read_pool_total + t.read_pool_total").Exec(edb).Error
+}
+
+func (edb *EventDb) updateUserWritePoolTotal(wpls []WritePoolLock, shouldIncrease bool) error {
+	var ids []string
+	var writepools []int64
+	for _, wpl := range wpls {
+		ids = append(ids, wpl.Client)
+		if shouldIncrease {
+			writepools = append(writepools, wpl.Amount)
+			continue
+		}
+		writepools = append(writepools, -wpl.Amount)
+
+	}
+
+	return CreateBuilder("users", "user_id", ids).
+		AddUpdate("write_pool_total", writepools, "users.write_pool_total + t.write_pool_total").Exec(edb).Error
+}
+
+func (edb *EventDb) updateUserPayedFees(users []User) error {
+	var ids []string
+	var fees []int64
+	for _, u := range users {
+		ids = append(ids, u.UserID)
+		fees = append(fees, u.PayedFees)
+	}
+
+	return CreateBuilder("users", "user_id", ids).
+		AddUpdate("payed_fees", fees, "users.payed_fees + t.payed_fees").Exec(edb).Error
+}
+
+func mergeUpdateUserCollectedRewardsEvents() *eventsMergerImpl[User] {
+	return newEventsMerger[User](TagUpdateUserCollectedRewards, withUniqueEventOverwrite())
+}
+
+func mergeUpdateUserTotalStakeEvents() *eventsMergerImpl[DelegatePoolLock] {
+	return newEventsMerger[DelegatePoolLock](TagLockStakePool, withUniqueEventOverwrite())
```

Comment:
This `withUniqueEventOverwrite()` will overwrite the previous one if there are more than one event from the same client, but the `TagLockStakePool` gives the `delta` value, we need to accumulate all of them. Please check https://github.com/0chain/0chain/blob/72a87c194322d79fafa7189cf99de7e210f2ec74/code/go/0chain.net/smartcontract/dbs/event/allocation.go#L249-L258 to see how to add up the values. And please also check other tags.